haystack/pipelines/utils.py

```diff
@@ -250,20 +254,24 @@ def print_eval_report(
         n_wrong_examples=n_wrong_examples,
         document_scope=document_scope,
         answer_scope=answer_scope,
+        wrong_examples_filter=wrong_examples_filter,
+        max_characters_per_wrong_examples_report=max_characters_per_wrong_examples_report,
     )
 
     print(f"{pipeline_overview}\n" f"{wrong_examples_report}")
 
 
-def _format_document_answer(document_or_answer: dict):
-    return "\n \t".join(f"{name}: {value}" for name, value in document_or_answer.items())
+def _format_document_answer(document_or_answer: dict, field_filter: List[str] = None):
+    if field_filter is None or len(field_filter) == 0:
+        field_filter = document_or_answer.keys()  # type: ignore
+    return "\n \t".join(f"{name}: {value}" for name, value in document_or_answer.items() if name in field_filter)  # type: ignore
 
 
-def _format_wrong_example(query: dict):
+def _format_wrong_example(query: dict, field_filter: List[str] = None):
     metrics = "\n \t".join(f"{name}: {value}" for name, value in query["metrics"].items())
-    documents = "\n\n \t".join(map(_format_document_answer, query.get("documents", [])))
+    documents = "\n\n \t".join([_format_document_answer(doc, field_filter) for doc in query.get("documents", [])])
```

Comment:
the squared brackets should not be needed here. Same with answers below...

Reply:
@tstadel it does because when we use the map function the second parameter (field_filter) is iterated as well causing havoc. That's why I had to convert the map into a list comprehension.

Reply:
Sure, you have to get rid of the `map()`. But `str.join` should be fine with generators as arg as well. So you can replace `[]` with `()` and `()` can be left out when generators are used as a function param. I'm pretty sure about that, but maybe I'm missing something...

Reply:
Aha cool, didn't know that. I'll try it out